### PR TITLE
Added the get_unanswered_question & add_answered_question API

### PR DESCRIPTION
**Pull request by rapirent**
Added the question API for the app frontend

### CODE REVIEW

File: smart_campus/app/views.py [hunk 1]
```diff
@@ -1359,3 +1361,56 @@ def group_delete_page(request, pk):
...
+
+    if not user or not station:
+        return HttpResponse('Either user or station does not exist', status=400)
+
+    unanswered_questions = Question.objects.exclude(user__pk=email).filter(
+        linked_station=station
+    )
+
+    random_index = random.randint(0, unanswered_questions.count() - 1)
+    random_unanswered_question = unanswered_questions[random_index]
```
Comment (Lee-W) [inline smart_campus/app/views.py]: If my memory serves me right, you could do it using `random.sample`

File: smart_campus/app/views.py [hunk 2]
```diff
@@ -1359,3 +1361,56 @@ def group_delete_page(request, pk):
...
+
+    choices = []
+    for i, choice in enumerate(Choice.objects.filter(question=random_unanswered_question)):
+        choices.append(choice.content)
+        if choice.is_answer:
+            index = i + 1
```
Comment (Lee-W) [inline smart_campus/app/views.py]: What is this index for?

File: smart_campus/app/views.py [hunk 3]
```diff
@@ -1359,3 +1361,56 @@ def group_delete_page(request, pk):
...
+        },
+        status=200
+    )
+
+
+@require_POST
+@csrf_exempt
+def add_answered_question(request):
+    question_id = request.POST.get('question_id')
+    email = request.POST.get('email')
```
Comment (Lee-W) [inline smart_campus/app/views.py]: I suggest that you change the key to `user_email` for readability.

Comment (rapirent) [reply smart_campus/app/views.py]: It sounds great, I will rename them.

File: smart_campus/app/views.py (hunk 2 above)
Comment (rapirent) [reply smart_campus/app/views.py]: This index is provide the app frontend to figure out which choice in the choices list is the answer.

File: smart_campus/app/views.py (hunk 1 above)
Comment (rapirent) [reply smart_campus/app/views.py]: Yeah, but making a query with model in django should not privide a list type which random.sample will return.
I'm curious if there is a much better way.

**Pull request merged**